src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs

```diff
@@ -162,25 +168,44 @@ private static ProcessModuleCollection GetModules(int processId, bool firstModul
                             BaseAddress = ntModuleInfo.BaseOfDll
                         };
 
-                        int length = Interop.Kernel32.GetModuleBaseName(processHandle, moduleHandle, chars, chars.Length);
+                        int length = 0;
+                        while ((length = Interop.Kernel32.GetModuleBaseName(processHandle, moduleHandle, chars, chars.Length)) == chars.Length)
+                        {
+                            char[] toReturn = chars;
+                            chars = ArrayPool<char>.Shared.Rent(length * 2);
```

Comment:
Why is it better to rent a new array first before returning the old one?

Reply:
The order of renting and returning doesn't matter.  But what does matter is that the variable holding on to the array being returned isn't the same one that's used for returning in the finally block. If it is,  and the Return call throws after having stored the array,  it could double return.